mtags/src/main/scala-3/scala/meta/internal/pc/CompletionItemResolver.scala

```diff
@@ -0,0 +1,88 @@
+package scala.meta.internal.pc
+
+import scala.meta.internal.mtags.MtagsEnrichments.*
+import scala.meta.pc.PresentationCompilerConfig
+import scala.meta.pc.SymbolDocumentation
+import scala.meta.pc.SymbolSearch
+
+import dotty.tools.dotc.core.Contexts.Context
+import dotty.tools.dotc.core.Flags.*
+import dotty.tools.dotc.core.Symbols.*
+import dotty.tools.dotc.core.Types.TermRef
+import org.eclipse.lsp4j.CompletionItem
+
+object CompletionItemResolver extends ItemResolver:
+
+  override def adjustIndexOfJavaParams = 0
+
+  def resolve(
+      item: CompletionItem,
+      msym: String,
+      search: SymbolSearch,
+      metalsConfig: PresentationCompilerConfig
+  )(using Context): CompletionItem =
+    SemanticdbSymbols.inverseSemanticdbSymbol(msym) match
+      case gsym :: _ if gsym != NoSymbol =>
+        search
+          .symbolDocumentation(gsym)
+          .orElse(
+            search.symbolDocumentation(gsym.companion)
+          ) match
+          case Some(info) if item.getDetail != null =>
+            enrichDocs(
+              item,
+              info,
+              metalsConfig,
+              fullDocstring(gsym, search),
+              gsym.is(JavaDefined)
+            )
+          case _ =>
+            item
+        end match
+
+      case _ => item
+    end match
+  end resolve
+
+  private def fullDocstring(gsym: Symbol, search: SymbolSearch)(using
```

Comment:
Not sure if we can sensibly extract this, so I left this method separate for Scala 2 and Scala 3 implementation.